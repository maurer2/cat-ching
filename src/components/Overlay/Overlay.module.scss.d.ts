import globalClassNames from '../../style.d';
declare const classNames: typeof globalClassNames & {
  readonly overlay: 'overlay';
  readonly wrapper: 'wrapper';
  readonly overlayText: 'overlayText';
  readonly resetButton: 'resetButton';
};
export = classNames;
