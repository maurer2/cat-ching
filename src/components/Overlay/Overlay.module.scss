.overlay {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  backdrop-filter: blur(3px) grayscale(100%);
}

.overlayText {
  margin: 0;
  padding: 1rem;
  font-size: 2rem;
  background: white;
}
