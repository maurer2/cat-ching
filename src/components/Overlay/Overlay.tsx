import React from 'react';

import style from './Overlay.module.scss';
// import * as Types from './Overlay.types';

function Overlay(): JSX.Element {
  return (
    <aside className={style.overlay}>
      <p className={style.overlayText}>
        Purrfect!
      </p>
    </aside>
  );
}

export default Overlay;
