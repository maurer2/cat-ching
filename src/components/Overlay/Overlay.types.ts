export type OverlayProps = Record<string, never>;
