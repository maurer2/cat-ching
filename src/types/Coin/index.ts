export type { CoinData, Coin } from './Coin';
