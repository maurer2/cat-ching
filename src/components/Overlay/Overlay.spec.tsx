/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { vi } from 'vitest';

import Component from './Overlay';
import * as Types from './Overlay.types';

describe('AppContainer', () => {
  const onReset = vi.fn();

  const propsDefault: Types.OverlayProps = {
    onReset,
  };

  const setup = (props = {}) => render(<Component {...propsDefault} {...props} />);

  it('renders', () => {
    const { queryByTestId } = setup();

    expect(queryByTestId('overlay')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('has child elements', () => {
    const { getByRole, getByText } = setup();

    expect(getByText('Purrfect!')).toBeInTheDocument();
    expect(getByRole('button')).toBeInTheDocument();
  });

  it('triggers reset callback on click', async () => {
    const user = userEvent.setup();
    const { getByText } = setup();

    await user.click(getByText('Reset'));
    expect(onReset).toHaveBeenCalled();
  });
});
