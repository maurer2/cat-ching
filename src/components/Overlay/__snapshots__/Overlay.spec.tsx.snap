// Vitest Snapshot v1

exports[`AppContainer > matches snapshot 1`] = `
<div>
  <aside
    class="_overlay_lnffa_1"
    data-testid="overlay"
  >
    <div
      class="_wrapper_lnffa_17"
    >
      <p
        class="_overlayText_lnffa_25"
      >
        Purrfect!
      </p>
      <button
        class="_resetButton_lnffa_30"
        type="button"
      >
        Reset
      </button>
    </div>
  </aside>
</div>
`;
